File: assignment 3/c3/programs/3.c
#include <stdio.h>

void printMatrix(int matrix[][4], int row, int col)
{
    for (int i = 0; i < row; i++)
    {
        for (int j = 0; j < col; j++)
        {
            printf("%d\t", matrix[i][j]);
        }
        printf("\n");
    }
}

int main()
{
    int mat[4][4] = {{1, 2, 3, 4},
                     {5, 6, 7, 8},
                     {9, 10, 11, 12},
                     {13, 14, 15, 16}};

    printf("old matrix:\n");
    printMatrix(mat, 4, 4);

    int mat_new[4][4];
    int n = 0;
    for (int i = 4; i > 0; i--)
    {
        for (int j = 0; j < 4; j++)
        {
            mat_new[j][i - 1] = mat[n][j];
        }
        n++;
    }

    printf("new matrix:\n");
    printMatrix(mat_new, 4, 4);

    return 0;
}

File: assignment 3/python/programs/3.py
def printmatrix(matrix):
    row=len(matrix)
    col=len(matrix[0])
    for i in range(row):
        for j in range(col):
            print(matrix[i][j], end="\t")
        print()
mat=[[1,2,3,4],[5,6,7,8],[9,10,11,12],[13,14,15,16]]
print("old matrix:")
printmatrix(mat)
n=0
mat_new = [[0 for x in range(4)] for y in range(4)]
for i in range(4,0,-1):
    for j in range(4):
        mat_new[j][i-1]=mat[n][j]
    n=n+1
print("new matrix:")
printmatrix(mat_new)